core/templates/domain/collection/collection.model.ts

```diff
@@ -32,35 +32,40 @@ interface ExplorationIdToNodeIndexMap {
 }
 
 export interface CollectionBackendDict {
-  'id': string;
-  'title': string;
-  'objective': string;
-  'language_code': string;
-  'tags': string[];
-  'schema_version': number;
+  // On creating a new Collection, properties below are always
+  // initialized with null values. Null here implies that these properties
+  // are yet to be assigned. See function createEmptyCollection for reference.
+  'id': string | null;
+  'title': string | null;
+  'objective': string | null;
+  'language_code': string | null;
+  'tags': string[] | null;
+  'schema_version': number | null;
+  'category': string | null;
+  'version': number | null;
```

Comment:
This is a backend dict... can you check with vojta if sending null values is ok or not. In case its not, you should throw an error that the following properties are null and have to be filled before making a request.

Reply:
@srijanreddy98 when an empty collection is created these values are null initially (Confirmed with vojta)
![image](https://user-images.githubusercontent.com/60519303/122430306-1b5a5d00-cfb1-11eb-9f9f-8c7e74e46bcf.png)
The id and version need to be null but those shouldn't be empty strings as well ? We're passing ids before creating a collection see this https://github.com/oppia/oppia/blob/b59fc11db163f039d8ee7ef6f7d0184ca5856a95/core/templates/components/entity-creation-services/collection-creation.service.ts#L69

Reply:
Got it... thanks for the explanation!